### ISSUE TITLE
mk-node hangs when running a test

### ISSUE THREAD

**Issue by bassosimone** [measurement-kit/measurement-kit#1483]

**Comment by bassosimone** [comment]
Apparently:

```
$ git bisect good
0620f27f8d732aa5cfac932b64f24ee0c9847c46 is the first bad commit
commit 0620f27f8d732aa5cfac932b64f24ee0c9847c46
Author: Simone Basso <bassosimone@gmail.com>
Date:   Mon Oct 30 16:47:15 2017 +0100
[snip]
```

Unclear how.

**Comment by bassosimone** [comment]
Okay, okay, I apparently did something wrong in the bisect. The culprit is:

```
commit 190365d201666393e97aaeff6496b3b961117085 (HEAD)
Author: Simone Basso <bassosimone@gmail.com>
Date:   Mon Oct 30 18:27:30 2017 +0100

    reactor: correctly deal with exception in child thread (#1445)
    
    * reactor: deal with exceptions in child threads
    
    * fix(getaddrinfo_async): don't move reactor and logger
    
    * fix(nettests/base_test.cpp): cannot move runnable
    
    Otherwise, runnable->logger is a null pointer.
```

(I find it interesting when these things happen. I also must say that probably all of this is caused by abusing of shared pointers and it would probably be smarted to enforce single ownership as much as we can. I am already spending a lot of time thinking about how to improve the situation. It's actually of course not possible to have these improvements in for v0.8.0, but I hope we'll have more unique ownership in v0.9.0, either via true unique pointers or, when that is not possible, having weak ownership.)

**Comment by bassosimone** [comment]
Fixed by #1484. More analysis of what was happening in #1485.
